nncf/experimental/torch/nas/bootstrapNAS/search/supernet.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+from nncf.experimental.torch.nas.bootstrapNAS.training.model_creator_helpers import resume_compression_from_state
+from nncf.torch.checkpoint_loading import load_state
+from nncf.torch.model_creation import create_nncf_network
+
+
+class SuperNetwork:
```

Comment:
It's hard to review interface properly without knowing how and where it's going to be used.
Is it possible to see example of usage?

Reply:
Here is an [example](https://github.com/IntelLabs/Hardware-Aware-Automated-Machine-Learning/blob/main/examples/bootstrapNAS/external_search_resnet50_supernet.ipynb) of how this wrapper is used in one third-party solution to search for efficient subnetworks. Thanks!

Reply:
Please rename `SuperNetwork` to `TrainedSuperNet` in order to avoid confusion that it can be trained.

Reply:
@ljaljushkin Renamed. Thanks!